test/grove-base-fork/Curve.t.sol

```diff
@@ -0,0 +1,1337 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+import "./ForkTestBase.t.sol";
+
+import { ICurvePoolLike as ICurvePoolLikeLib } from "../../src/libraries/CurveLib.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+interface ICurvePoolLike is ICurvePoolLikeLib {
+    function calc_token_amount(uint256[] memory amounts, bool is_deposit) external view returns (uint256);
+    function calc_withdraw_one_coin(uint256 amount, int128 index) external view returns (uint256);
+    function get_dy(int128 i, int128 j, uint256 dx) external view returns (uint256);
+}
+
+contract MockCgUSD is ERC20 {
+    
+    uint8 private immutable _decimals;
+    
+    constructor() ERC20("cgUSD", "cgUSD") {
+        _decimals = 6;
+    }
+
+    function decimals() public view override returns (uint8) {
+        return _decimals;
+    }
+}
+
+contract CurveTestBase is ForkTestBase {
+
+    // TODO: replace with real target pool once available
+    address constant CURVE_POOL = 0xa9D6867C347B8b5f395B8421FB31710B8Fb21a16; // USDC/ cgUSD
+    address constant CGUSD = 0xCa72827a3D211CfD8F6b00Ac98824872b72CAb49;
+
+    IERC20 curveLp = IERC20(CURVE_POOL);
+    IERC20 cgUSD = IERC20(CGUSD);
+
+    ICurvePoolLike curvePool = ICurvePoolLike(CURVE_POOL);
+
+    bytes32 curveDepositKey;
+    bytes32 curveSwapKey;
+    bytes32 curveWithdrawKey;
+
+    uint256 maxSlippage;
+
+    function setUp() public virtual override  {
+        super.setUp();
+
+        // There is something weird with the CGUSD token that causes deal to under/overflow.
+        // Make a mock Decimal 6 token and replace it for these tests
+        uint256 curveCGUSDBalance = cgUSD.balanceOf(CURVE_POOL);
+        ERC20Mock mockCgUSD = ERC20Mock(address(new MockCgUSD()));
+        vm.etch(CGUSD, address(mockCgUSD).code);
+        // Preserve the balance of the CGUSD token in the pool
+        deal(address(cgUSD), CURVE_POOL, curveCGUSDBalance);
+
+        if (curveCGUSDBalance < 100_000_000e6 || usdcBase.balanceOf(CURVE_POOL) < 100_000_000e6) {
+            // boost liquidity to help later tests
+            uint256[] memory amounts = new uint256[](2);
+            amounts[0] = 100_000_000e6;
+            amounts[1] = 100_000_000e6;
```

Comment:
Are these ever intended to be different? If not then maybe no need to duplicate?

Reply:
I'm not sure I follow. We need the two values in an array, so I could do a double assignment but I don't love that formatting.